### ISSUE TITLE
Cannot find module 'electron-compile' / Error is in blueprint

### ISSUE THREAD

**Issue by gossi** [felixrieseberg/ember-electron#318]
Hey guys,

When I did `ember electron:package` and start the generated `.app` I do get the error `Cannot find module 'electron-compile'`. I wasn't quite sure, which `ember-electron` version I was on at that time, so I updated to the most up-to-date one `2.7.1` though the same error appeared. I tried running the blueprint again, no help either. I also tried different versions of `ember-electron` and `electron-forge`, so I switched back to `v2.7.1` and was playing around.

What keeps me wondering, is when I did (re)run the blueprint is what happens to my `package.json` and my `electron-forge-config.js`.

```js
// package.json
{
  // ..
  "config": {
    "forge": "./ember-electron/electron-forge-config.js"
  },
  // ..
}
```

After running the blueprint, I had my config replaced with the default electron-forge config object. Two things going wrong here:

1. It doesn't detect properly that I do have my forge config in a different file (although I read the code, which looks fine to me)
2. It doesn't update my existing config to changes that are required to solve my actualy problem I'm having.

I manually added `"packageManager": "yarn",` to my present `"electronPackagerConfig"` config. The next time I did run `ember electron:package` all my dependencies are correctly copied over to the generated `.app` which was then running fine.

I'm wondering wether it is required to add `"packageManager": "yarn",` at all. I think whenever `ember electron:*` is run, it should auto-detect which package manager is used, set it on the fly and proceed with whatever it is doing. Sounds like the better solution to me.

**Comment by anulman** [comment]
@gossi What version of e-electron are you using? This becomes part of the blueprint (i.e. initial generator scripts) as of 2.7.1: https://github.com/felixrieseberg/ember-electron/commit/fc149c05883e903ea897c2a35594dfdb65f24a7c

**Comment by gossi** [comment]
I'm on 2.7.1 - That's where I observed the symptions I described above.

**Comment by anulman** [comment]
Thanks @gossi. Had you run the generator scripts, i.e. `ember g ember-electron` after updating?

If so, we can debug further and figure out why it didn't run as expected. If not, thinking maybe we should flag this better in the release notes.

**Comment by gossi** [comment]
Yup, I run the generator again: `ember g ember-electron`. That's where the two things I pointed out above went wrong.

**Comment by hinti** [comment]
Maybe it's a node/npm related problem?

```electron:package --environment=production --platform win32```

**node 6.11.5 & npm 3.3.10**

Everything OK. App starts.

**node 8.9.0 & npm 5.5.1**

Trying to start the app:

![prob](https://user-images.githubusercontent.com/1922749/32319880-480ec75e-bfbc-11e7-87da-28b0511c65b7.png)

**Comment by anulman** [comment]
@gossi Sorry I missed your last comment. The version-related issue is likely related to the npm / yarn decision tree.

I know there were relevant fixes upstream. @bendemboski do you remember if this is handled in e-electron? Thinking maybe we need to move [this snippet in the blueprint that checks for yarn / sets packageManager](https://github.com/felixrieseberg/ember-electron/blob/master/blueprints/ember-electron/index.js#L132-L135) above [this snippet in the blueprint, which exits if forge config already exists](https://github.com/felixrieseberg/ember-electron/blob/master/blueprints/ember-electron/index.js#L114-L118)

**Comment by jacobq** [comment]
@gossi I started peeking through GH issues here with the intent of closing those that are no longer relevant (either fixed or no longer relevant). With Node 10.x becoming active LTS very soon I wanted to ask if this problem is still occurring or if there is otherwise any work that still needs to be done before closing this issue. (I don't have easy access to a Mac but have not seen similar problems in Linux or Windows packages for a long time.)

**Comment by gossi** [comment]
The last time I worked with e-electron everything was working for me. I guess what I described above was something about migrating to a new version, which was problematic and required manual steps. However, I close it, since it's kinda dated and if the problem comes up again, that person should create a new issue then.
